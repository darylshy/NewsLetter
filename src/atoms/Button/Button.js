import React from 'react';
import PropTypes from 'prop-types';
import classes from './Button.module.scss';

const Button = props => {
  return (
    <button
      className={classes.Button}
      onClick={props.handleClick}
      disabled={props.disabled}
    >
      {props.buttonText.toUpperCase()}
    </button>
  );
};

Button.propTypes = {
  buttonText: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool
};

export default Button;
