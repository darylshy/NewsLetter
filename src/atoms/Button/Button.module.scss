@import "../../styles/global/globals.scss";
.Button {
  background-color: $danger-red;
  border: none;
  border-radius: $border-radius;
  padding: 1.2rem 3.5rem;
  font-family: "Azo-Sans-Bold";
  color: #fff;
  @include desktop {
    font-size: 2rem;
  }
  @include tabletL {
    font-size: 2rem;
  }
  @include tabletP {
    font-size: 2rem;
  }
  @include mobile {
    font-size: 1.6rem;
    flex: 1;
  }
}
