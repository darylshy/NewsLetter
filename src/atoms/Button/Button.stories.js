import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean } from '@storybook/addon-knobs/react';

import Button from './Button';
storiesOf('Atoms/Button', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add(
    'Standard Form Button',
    () => (
      <Button
        handleClick={action('Button Clicked.')}
        buttonText={text('buttonText')}
        disabled={boolean('disabled')}
      />
    ),
    {
      info: {
        text: 'Standard form button'
      }
    }
  );
