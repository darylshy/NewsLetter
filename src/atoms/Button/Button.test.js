import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from './Button';
import React from 'react';

configure({ adapter: new Adapter() });

describe('<Button />', () => {
  let component;
  beforeEach(() => {
    component = shallow(<Button buttonText="Button" handleClick={mock} />);
  });
  const mock = jest.fn();
  it('should render an Button component', () => {
    expect(component).toMatchSnapshot();
  });
  it('should find a button', () => {
    expect(component.find('button')).toHaveLength(1);
  });
  it('should have correct button label', () => {
    expect(component.find('button').props().children).toBe('BUTTON');
  });
  it('should trigger callback on click', () => {
    component.find('button').simulate('click');
    expect(mock.mock.calls.length).toEqual(1);
  });
});
