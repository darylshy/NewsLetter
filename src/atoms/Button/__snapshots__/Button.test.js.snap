// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render an Button component 1`] = `ShallowWrapper {}`;
